liquidity_pool/src/liquidity.rs

```diff
@@ -120,16 +126,22 @@ pub trait LiquidityModule:
         );
 
         let new_nonce = self.mint_position_tokens(&borrow_token_id, &borrow_amount_in_tokens);
-        let timestamp = self.blockchain().get_block_timestamp();
+        let round_no = self.blockchain().get_block_round();
         let borrow_position = BorrowPosition::new(
-            timestamp,
+            round_no,
             lend_tokens,
             borrow_amount_in_tokens.clone(),
             payment_lend_token_id,
+            self.borrow_index().get(),
         );
 
         self.borrow_position(new_nonce).set(&borrow_position);
 
+        self.update_borrow_index(self.borrow_index_last_used().get());
+        let rewards_increase = self.update_rewards_reserves(self.borrow_index_last_used().get());
+        self.update_supply_index(rewards_increase);
+        self.update_index_last_used();
```

Comment:
I agree with this. Also, since the index updates are done only once per round it makes sense for the logic to be grouped and done at the beginning. Otherwise the same user might get different results if, for example, he is the first user to call a method at the start of the round or not, as the index updates use `delta_rounds` which is 0 if someone called any other method before.

Additional suggestions - after moving this logic into a single method:
- for the `update_borrow_index` and `update_rewards_reserves` methods, extract the `let delta_rounds = self.get_round_diff(borrow_index_last_used);` and provide it as an argument, since it's common for both methods
- add a check for `delta_rounds > 0`, otherwise it doesn't make sense re-compute the rewards or the indices, nor to overwrite the storage with the same values - it should save some gas for most users

Reply:
Yep, it makes sense to create a method for the index updates.

I am hesitant to put the method at the start of the endpoint. There are some checks at the beginning that may end up not going through with the endpoint. In this case, the computation done for updating the indexes is useless.
I think I'm going to put this method after the checks at the start of the endpoint.